field/src/field.rs

```diff
@@ -479,8 +478,8 @@ impl<F: PrimeCharacteristicRing> BasedVectorSpace<F> for F {
     }
 
     #[inline]
-    fn from_basis_coefficients_iter<I: Iterator<Item = F>>(mut iter: I) -> Self {
-        iter.next().unwrap()
+    fn from_basis_coefficients_iter<I: ExactSizeIterator<Item = F>>(mut iter: I) -> Option<Self> {
```

Comment:
Hm it seems like equivalent impls could be written without needing sizes upfront, albeit with a bit more annoying code. I guess `ExactSizeIterator` should fine though since it's probably only used internally